middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The destructuring of `userId` and `channelId` from the request object doesn't handle the case where `req` is undefined. This could lead to a TypeError if `req` is not provided. It would be better to add a default value for `req` in the destructuring assignment.
```diff
-  const {userId} = req || {};
-  const {channelId} = req?.query || {};
+  const {userId} = req || {};
+  const {channelId} = (req || {}).query || {};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->